### ISSUE TITLE
Paste still fucked with on spicejet.com 

### ISSUE THREAD

**Issue by quinncomendant** [jswanner/DontFuckWithPaste#16]
On the [payment screen](https://book.spicejet.com/Payment.aspx) of the flight booking flow at SpiceJet, the credit card number field doesn't allow paste using command-v on a Mac, although it does allow right-click>paste.

![screen shot 2016-10-24 at 18 42 18](https://cloud.githubusercontent.com/assets/6884276/19668979/388dd5d4-9a21-11e6-9768-a2a5fca9139e.png)

**Comment by jswanner** [comment]
I looked at spicejet.com, I didn't make it as far as the payment page, but I did notice similar behavior on an earlier page, where they ask for a phone number.

And, at least with the phone number field, they did not prevent the `paste` event (as you noted, right-click & paste works), but instead are hooking into the `keydown` event. For those fields, the only keys they are allowing to be used are: numbers, hyphen, arrows, tab, backspace, and delete.

So, ⌘v isn't working because they are telling the browser not to recognize the "⌘" then the "v" keys.

This extension works by preventing a site's JavaScript code from ever receiving `copy` or `paste` events, thereby not letting them block those events. I'm not comfortable with doing the same for `keydown`, `keyup`, and `keypress` events.
